Assets/Scripts/MenuLayerSystem/InteractableManager.cs

```diff
@@ -0,0 +1,54 @@
+using System.Collections.Generic;
+using UnityEngine;
+
+public class InteractableManager : MonoBehaviour
+{
+    public List<IInteractable> Interactables { get; private set; }
+
+    private void Awake()
+    {
+        Interactables = new List<IInteractable>();
+    }
+
+    public void AddInteractableToList(IInteractable interactable) => Interactables.Add(interactable);
+
+    /// <summary>
+    /// This function will put the interactable element in the first position on the list and then reorder the list
+    /// </summary>
+    /// <param name="interactable"> The interactable element that needs to be put first</param>
+    public void PutElementToFirstInList(IInteractable interactable)
+    {
+        List<IInteractable> tempList = new List<IInteractable>();
+
+        tempList.Add(interactable);
+
+        foreach (IInteractable iInteractable in Interactables)
```

Comment:
So everytime you add an element to the list, you check for duplicates?
Isn't using the `List<T>.Contains(T)` easier?